tests/integration/profile_test.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import os
+
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+
+from .testlib import cmd as libcmd
+
+
+NMCLI_CONADD_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'dummy',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    'dummy0',
+]
+
+PROFILE_DIRECTORY = '/etc/NetworkManager/system-connections/'
+
+
+def test_delete_inactive_profiles():
+    libcmd.exec_cmd(NMCLI_CONADD_CMD)
+    profile_exists = os.path.isfile(
+        PROFILE_DIRECTORY + 'testProfile.nmconnection'
+    )
+    assert profile_exists
+
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: 'dummy0',
+                Interface.TYPE: InterfaceType.DUMMY,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    desired_state[Interface.KEY][0][Interface.STATE] = InterfaceState.DOWN
+    libnmstate.apply(desired_state)
```

Comment:
This should be part of the nmstate iface creation in a contextmanager:
```
with dummy_interface(...):
    assert not _profile_exists()
```
You need to assert that the inactive profile is not there anymore after defining the dummy using nmstate, not after setting it `down`.

Reply:
Fixed. Thanks!